HISTORY.md

```diff
@@ -3,35 +3,41 @@
 
 This incorporates all changes after 1.3.5 up to 1.3.6.
 
+* deps: `path-to-regexp@6.2.0` (See [release changelog](https://github.com/pillarjs/path-to-regexp/releases/tag/v6.2.0))
+  - Support for **named capturing groups** in paths using `RegExp`, i.e.: `/\/(?<group>.+)/` would result in `req.params.group` being populated.
+  - Custom **prefix and suffix groups** using `{}`, e.g.: `/:entity{-:action}?` would match `/user` and `/user-delete`.
+  - **BREAKING**: Parenthesis and bracket literals now require escaping, i.e.: `/user(:id)` or `/user{:id}` no longer match `/user(42)`, `/user{42}` as before unless written as `/user\\(:id\\)` or `/user\\{:id\\}`.
```

Comment:
I didn't test the `{}` one, since that was added in path-to-regexp 6, but I tested the `/user(:id)` one and `/user(:id)` doesn't match `/user(42)` in 2.0.0-beta.1 either, and needs the escapes. Was this change just for `{}` and was the `()` a typo? I cannot find this noted in `path-to-regexp` changelog, either.

Reply:
I went ahead and tested `/user{:id}` in 2.0.0-beta.1 as well and it doesn't work with or without `\\` chars. Here is the test I was using to try it out:
```js
it('should match single path segment', function (done) {
  var router = new Router()
  var route = router.route('/user(:id)')
  var server = createServer(router)

  route.all(sendParams)

  request(server)
    .get('/user(1)')
    .expect(200, '{"id":"1"}', done)
})
```
It returns 404 for both 2.0.0-beta.1 release and on this PR. I'll add these as tests to the test suite as well, as they are good tests to have 👍

Reply:
I tested this using `path-to-regexp@6.2.0` itself, if I remember correctly. Yeah, without escaping brackets you'd get a `404`.

```js
var match = require("path-to-regexp").match

match('/user/\\(:id\\)')('/user/(42)')
// {path: "/user/(42)", index: 0, params: {id: "42"}}

m('/user/\\{:id\\}')('/user/{42}')
// {path: "/user/{42}", index: 0, params: {id: "42"}}
```

Reply:
Right, I understand that. But I guess I wasn't clear in my point: I am saying that I don't think that _changed_ between 2.0.0-beta.1 and this PR, so it doesn't seem like it should be in our History as a breaking change, since nothing broke. For example, using the path-to-regexp that is in 2.0.0-beta.1 directly, as you tested directly with the one in this PR, here are the results on your tests:
```
$ node
Welcome to Node.js v16.7.0.
Type ".help" for more information.
> require("path-to-regexp/package").version
'3.2.0'
> match=require("path-to-regexp").match
[Function: match]
> match('/user/(:id)')('/user/(42)')
false
> match('/user/\\(:id\\)')('/user/(42)')
{ path: '/user/(42)', index: 0, params: { id: '42' } }
> match('/user/{:id}')('/user/{42}')
{ path: '/user/{42}', index: 0, params: { id: '42' } }
> match('/user/\\{:id\\}')('/user/{42}')
{ path: '/user/{42}', index: 0, params: { id: '42' } }
```

You can see that yes, it changes the `{:id}` one, but as I was saying, it doesn't seem to change the parenthesis one at all, right? So when it says "`/user(:id)` [...] no longer match `/user(42)`" in this update, how is that breaking, as it didn't work in the previous version, either.

Reply:
Ah, gotcha 👍🏼. Sorry, misunderstood what you were trying to say. Yeah, you're right. Maybe we could move that item about escaping brackets to the previous release changelog? I don't see it mentioned anywhere else and I think it should be.

Reply:
Yes, if indeed it does work in router 1.x and then didn't work in router 2.0.0-beta.1 then it was a miss for that changelog and we should add it there. I haven't confirmed it as I am on mobile now, but if confirmed, one of us can update it.

Reply:
Looks like it did work in `path-to-regexp@0.1.7`.

```js
var pathToRegexp = require('path-to-regexp@0.1.7')
'/user/(42)'.match(pathToRegexp('/user/(:id)'))

// ['/user/(42)', '(42)']
```

Reply:
Ok, cool. So then the entry does not belong in this change set, as it's just a missing one from 2.0.0-beta.1. It should at least be removed from this changeset. Ideally we should add it over retroactively to the 2.0.0-beta.1 changeset, but you don't need to do that in this PR if you don't want to.